tests/test_cost_review_server.py

```
@@ -56,9 +61,98 @@ def test_healthz_is_unauthenticated_and_ok() -> None:
+        status, body = _request("GET", f"{base}/api/v1/workflow_runs", token)
+        assert status == 200
+        assert body["items"] == []
+    finally:
+        server.shutdown()
+
+
+def test_trace_read_defaults_cannot_bypass_trace_purpose_authorization() -> None:
+    """Default trace exposure still requires the separate trace purpose scope."""
+    security = SecurityConfig(
+        auth_token="",
+        admin_token="admin_secret",
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to argument: "admin_token"

(S106)